src/app/modules/reef/pages/card-admin-page/card-admin-page.ts

```diff
@@ -0,0 +1,304 @@
+import {TokenUtil} from 'src/app/shared/utils/token.util';
+import {countryCodes} from '../../../../shared/data/country_codes';
+import {CountryISO, PhoneNumberFormat, SearchCountryField,} from 'ngx-intl-tel-input';
+import {NgForm} from '@angular/forms';
+
+@Component({
+  selector: 'app-card-admin-page',
+  templateUrl: './card-admin-page.html',
+  styleUrls: ['./card-admin-page.scss'],
+})
+export class CardAdminPage extends NgDestroyableComponent implements OnDestroy {
+  set supportedTokens(val: { tokenSymbol: TokenSymbol; src: string }[]) {
+    this.supportedTokensSub.next(val);
+    this.selTokenSub.next(val[0].tokenSymbol);
+  }
+
+  baanxBaseUrl = 'https://reeffinance.baanx.co.uk';
+  cardBaseUrl = environment.reefNodeApiUrl;
+  accountCardRegistered$: Observable<boolean>;
+  cardVerified$: Observable<boolean>;
+  iframeSession$: Observable<any>;
+  createUser: Subject<any> = new Subject<any>();
+  destroyed$ = new Subject<void>();
+  @ViewChild('cardFormDialog') cardFormDialog: TemplateRef<any>;
+  loading$: Observable<boolean>;
+
+  selectedTokenBalance$: Observable<Token>;
+  supportedTokensSub = new BehaviorSubject<{ tokenSymbol: TokenSymbol; src: string }[]>([]);
+  selTokenSub = new ReplaySubject<TokenSymbol>();
+  tokenAmountSub = new BehaviorSubject<number>(null);
+  TokenSymbol = TokenSymbol;
+  cardBalance$: Observable<string>;
+  TokenUtil = TokenUtil;
+  GENDER_OPTIONS: { v: string; l: string }[] = [
+    {v: 'M', l: 'Male'},
+    {v: 'F', l: 'Female'},
+  ];
+  countryCodes = countryCodes;
+  SearchCountryField = SearchCountryField;
+  CountryISO = CountryISO;
+  PhoneNumberFormat = PhoneNumberFormat;
+  preferredCountries: CountryISO[] = [
+    CountryISO.UnitedStates,
+    CountryISO.Turkey,
+    CountryISO.Brazil,
+    CountryISO.Italy,
+    CountryISO.Vietnam,
+    CountryISO.UnitedKingdom,
+    CountryISO.India,
+    CountryISO.Russia,
+    CountryISO.Indonesia,
+    CountryISO.Germany,
+    CountryISO.Australia,
+    CountryISO.Indonesia,
+    CountryISO.Spain,
+    CountryISO.France,
+    CountryISO.Canada,
+    CountryISO.Thailand,
+    CountryISO.Japan,
+    CountryISO.Argentina,
+    CountryISO.Ukraine,
+    CountryISO.Malaysia,
+    CountryISO.Poland,
+    CountryISO.Romania,
+    CountryISO.China,
+    CountryISO.Singapore,
+    CountryISO.Switzerland,
+    CountryISO.Portugal,
+  ];
+  displayFormNotValid: boolean;
+
+  constructor(
+    public readonly connectorService: ConnectorService,
+    private dialog: MatDialog,
+    private notificationService: NotificationService,
+    private ethAuthService: EthAuthService,
+    private http: HttpClient,
+    private tokenBalanceService: TokenBalanceService
+  ) {
+    super();
+    this.supportedTokens = TokenBalanceService.SUPPORTED_CARD_TOKENS;
+    const existingUser$ = connectorService.providerUserInfo$.pipe(
+      switchMap((userInfo: any) => {
+        if (userInfo) {
+          return this.http
```

Comment:
can we have services do the actual http requests using the `HttpClient`, as we do in the whole app? This way the component will be much cleaner instead of it being 300 lines